### ISSUE TITLE
No connection to SAMBA4 AD because of missing cert

### ISSUE THREAD

**Issue by zefanja** [linuxmuster/linuxmuster-base7#52]
I'm not sure if it is a `linuxmuster-setup` issue, but the inital setup does not finish, because of this error:

```
[2018/12/04 14:10:16.281877,  0] ../source4/lib/tls/tlscert.c:57(tls_cert_generate)
samba: setproctitle not initialized, please either call setproctitle_init() or link against libbsd-ctor.
samba: setproctitle not initialized, please either call setproctitle_init() or link against libbsd-ctor.
  TLS autogeneration skipped - some TLS files already exist
[2018/12/04 14:10:16.290693,  0] ../source4/lib/tls/tls_tstream.c:1232(tstream_tls_params_server)
  TLS failed to initialise cafile /etc/linuxmuster/ssl/cacert.pem - Error while reading file.
[2018/12/04 14:10:16.290876,  0] ../source4/ldap_server/ldap_server.c:1153(ldapsrv_task_init)
  ldapsrv failed tstream_tls_params_server - NT_STATUS_CANT_ACCESS_DOMAIN_INFO
[2018/12/04 14:10:16.290986,  0] ../source4/smbd/service_task.c:35(task_server_terminate)
  task_server_terminate: [Failed to startup ldap server task]
```

in `setup.final.log` I got this:

```
linuxmuster-import-devices                                           ####
Command line::
Option json is a modifier option
Option verbose is a modifier option
Hmmh. do not know what to do with option info
* forcing info mode
Option combinations successfully checked
#### No connection to Samba4 AD!                                          ####
Calling console printout
ERROR   -1: No connection to Samba4 AD!
```

These are the file I have in `/etc/linuxmuster/ssl`:

```
ls -al /etc/linuxmuster/ssl/
total 40
drwxr-x--- 2 root ssl-cert 4096 Dez  4 14:13 .
drwxr-xr-x 7 root root     4096 Dez  4 14:09 ..
-rw-r----- 1 root ssl-cert    0 Dez  4 14:09 cacert.pem.b64
-rw------- 1 root ssl-cert 1766 Dez  4 14:09 cakey.pem
-rw------- 1 root ssl-cert 1679 Dez  4 14:09 docker.key.pem
-rw-r----- 1 root ssl-cert    0 Dez  4 14:09 firewall.cert.pem.b64
-rw-r----- 1 root ssl-cert    0 Dez  4 14:09 firewall.fullchain.pem
-rw------- 1 root ssl-cert 1675 Dez  4 14:09 firewall.key.pem
-rw------- 1 root ssl-cert 2236 Dez  4 14:09 firewall.key.pem.b64
-rw-rw-rw- 1 root root     1679 Dez  4 14:13 lmn1.cert.bundle.pem
-rw------- 1 root ssl-cert 1679 Dez  4 14:09 lmn1.key.pem
-rw------- 1 root ssl-cert 1675 Dez  4 14:09 mail.key.pem
-rw------- 1 root ssl-cert 1679 Dez  4 14:09 opsi.key.pem
```

This happened after the initial setup with the latest appliance.

**Comment by HappyBasher** [comment]
There is no firewall.cert.pem. Please post version of linuxmuster-base7 and content of /var/log/linuxmuster/setup.ssl.log.
Is this reproducible?

**Comment by zefanja** [comment]
Version is 7.0.22. I did the setup with the webUI. I guess the firewall, opsi certs were created by the webUI (although I don't have an OPSI host and skipped firewall setup).

```
root@lmn1:~# cat /var/log/linuxmuster/setup.ssl.log 
------------------------------------------------------------------------------
#### 2018-12-04 14:09:57                                                  ####
#### chmod 400 /etc/linuxmuster/.secret/cakey                             ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:57                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/cakey.pem -aes128 -passout pass:g3TAllJCijyTN7nB 2048 ####
Generating RSA private key, 2048 bit long modulus
...................................................................+++
.........+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:57                                                  ####
#### openssl req -batch -x509 -subj /O=*************************************/OU=INTERNAL/CN=INTERNAL.************ /subjectAltName=INTERNAL.************ / -new -nodes  -passin pass:g3TAllJCijyTN7nB -key /etc/linuxmuster/ssl/cakey.pem -sha256 -days 3650 -out /etc/linuxmuster/ssl/cacert.pem ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:57                                                  ####
#### openssl x509 -in /etc/linuxmuster/ssl/cacert.pem -inform PEM -out /etc/linuxmuster/ssl/cacert.crt ####
x509: Cannot open input file /etc/linuxmuster/ssl/cacert.pem, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:57                                                  ####
#### ln -sf /etc/linuxmuster/ssl/cacert.crt /usr/local/share/ca-certificates/linuxmuster_cacert.crt ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### update-ca-certificates                                               ####
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### base64 /etc/linuxmuster/ssl/cacert.pem > /etc/linuxmuster/ssl/cacert.pem.b64 ####
base64: /etc/linuxmuster/ssl/cacert.pem: No such file or directory
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/lmn1.key.pem 2048           ####
Generating RSA private key, 2048 bit long modulus
.....................................................................................................+++
...........+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl req -batch -subj /O=*************************************/OU=INTERNAL/CN=lmn1.internal.************ /subjectAltName=lmn1.internal.************ / -new -key /etc/linuxmuster/ssl/lmn1.key.pem -out /etc/linuxmuster/ssl/lmn1.csr ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl x509 -req -in /etc/linuxmuster/ssl/lmn1.csr -CA /etc/linuxmuster/ssl/cacert.pem -passin pass:g3TAllJCijyTN7nB -CAkey /etc/linuxmuster/ssl/cakey.pem -CAcreateserial -out /etc/linuxmuster/ssl/lmn1.cert.pem -sha256 -days 3650 ####
x509: Cannot open input file /etc/linuxmuster/ssl/lmn1.csr, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### ln -sf /etc/linuxmuster/ssl/lmn1.cert.pem /etc/cups/ssl/server.crt   ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### ln -sf /etc/linuxmuster/ssl/lmn1.key.pem /etc/cups/ssl/server.key    ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### service cups restart                                                 ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/firewall.key.pem 2048       ####
Generating RSA private key, 2048 bit long modulus
.....................................................................................................................+++
....................+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl req -batch -subj /O=*************************************/OU=INTERNAL/CN=firewall.internal.************ /subjectAltName=firewall.internal.************ / -new -key /etc/linuxmuster/ssl/firewall.key.pem -out /etc/linuxmuster/ssl/firewall.csr ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl x509 -req -in /etc/linuxmuster/ssl/firewall.csr -CA /etc/linuxmuster/ssl/cacert.pem -passin pass:g3TAllJCijyTN7nB -CAkey /etc/linuxmuster/ssl/cakey.pem -CAcreateserial -out /etc/linuxmuster/ssl/firewall.cert.pem -sha256 -days 3650 ####
x509: Cannot open input file /etc/linuxmuster/ssl/firewall.csr, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### base64 /etc/linuxmuster/ssl/firewall.key.pem > /etc/linuxmuster/ssl/firewall.key.pem.b64 ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### base64 /etc/linuxmuster/ssl/firewall.cert.pem > /etc/linuxmuster/ssl/firewall.cert.pem.b64 ####
base64: /etc/linuxmuster/ssl/firewall.cert.pem: No such file or directory
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/mail.key.pem 2048           ####
Generating RSA private key, 2048 bit long modulus
.........................................................+++
................................................................+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl req -batch -subj /O=*************************************/OU=INTERNAL/CN=mail.internal.************ /subjectAltName=mail.internal.************ / -new -key /etc/linuxmuster/ssl/mail.key.pem -out /etc/linuxmuster/ssl/mail.csr ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl x509 -req -in /etc/linuxmuster/ssl/mail.csr -CA /etc/linuxmuster/ssl/cacert.pem -passin pass:g3TAllJCijyTN7nB -CAkey /etc/linuxmuster/ssl/cakey.pem -CAcreateserial -out /etc/linuxmuster/ssl/mail.cert.pem -sha256 -days 3650 ####
x509: Cannot open input file /etc/linuxmuster/ssl/mail.csr, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/docker.key.pem 2048         ####
Generating RSA private key, 2048 bit long modulus
...........+++
....................................+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl req -batch -subj /O=*************************************/OU=INTERNAL/CN=docker.internal.************ /subjectAltName=docker.internal.************ / -new -key /etc/linuxmuster/ssl/docker.key.pem -out /etc/linuxmuster/ssl/docker.csr ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:58                                                  ####
#### openssl x509 -req -in /etc/linuxmuster/ssl/docker.csr -CA /etc/linuxmuster/ssl/cacert.pem -passin pass:g3TAllJCijyTN7nB -CAkey /etc/linuxmuster/ssl/cakey.pem -CAcreateserial -out /etc/linuxmuster/ssl/docker.cert.pem -sha256 -days 3650 ####
x509: Cannot open input file /etc/linuxmuster/ssl/docker.csr, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### openssl genrsa -out /etc/linuxmuster/ssl/opsi.key.pem 2048           ####
Generating RSA private key, 2048 bit long modulus
........+++
....................................................................................................+++
unable to write 'random state'
e is 65537 (0x010001)
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### openssl req -batch -subj /O=*************************************/OU=INTERNAL/CN=opsi.internal.************ /subjectAltName=opsi.internal.************ / -new -key /etc/linuxmuster/ssl/opsi.key.pem -out /etc/linuxmuster/ssl/opsi.csr ####
req: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### openssl x509 -req -in /etc/linuxmuster/ssl/opsi.csr -CA /etc/linuxmuster/ssl/cacert.pem -passin pass:g3TAllJCijyTN7nB -CAkey /etc/linuxmuster/ssl/cakey.pem -CAcreateserial -out /etc/linuxmuster/ssl/opsi.cert.pem -sha256 -days 3650 ####
x509: Cannot open input file /etc/linuxmuster/ssl/opsi.csr, No such file or directory
x509: Use -help for summary.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### cat /etc/linuxmuster/ssl/firewall.cert.pem /etc/linuxmuster/ssl/cacert.pem > /etc/linuxmuster/ssl/firewall.fullchain.pem ####
cat: /etc/linuxmuster/ssl/firewall.cert.pem: No such file or directory
cat: /etc/linuxmuster/ssl/cacert.pem: No such file or directory
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### mkdir -p /var/lib/samba/sysvol/internal.************ /tls               ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### cp /etc/linuxmuster/ssl/cacert.pem /var/lib/samba/sysvol/internal.************ /tls/cacert.pem ####
cp: cannot stat '/etc/linuxmuster/ssl/cacert.pem': No such file or directory
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### chgrp -R ssl-cert /etc/linuxmuster/ssl                               ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### chmod 750 /etc/linuxmuster/ssl                                       ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### chmod 640 /etc/linuxmuster/ssl/*                                     ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-12-04 14:09:59                                                  ####
#### chmod 600 /etc/linuxmuster/ssl/*key*                                 ####
------------------------------------------------------------------------------
```

**Comment by zefanja** [comment]
I guess I found the issue. If the school name has spaces in its name, the creation of the root CA cert fails.

```
root@lmn1:~# openssl req -batch -x509 -subj /O=School Name/OU=INTERNAL/CN=INTERNAL.EXAMPLE.COM/subjectAltName=INTERNAL.CDSC.AC.TH/ -new -nodes  -passin pass:MkkZGiw0tPnHLlaV -key /etc/linuxmuster/ssl/cakey.pem -sha256 -days 3650 -out /etc/linuxmuster/ssl/cacert.pem
req: Use -help for summary.
```

**Comment by HappyBasher** [comment]
Fixed in [7.0.28](https://github.com/linuxmuster/linuxmuster-base7/tree/7.0.28).
